front/pages/api/w/[wId]/data_sources/[name]/configuration.ts

```diff
@@ -0,0 +1,164 @@
+    WithAPIErrorReponse<PostDataSourceConfigurationResBody | void>
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  if (!auth.isBuilder()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "data_source_auth_error",
+        message:
+          "Only the users that are `builders` for the current workspace can access Data Source configuration.",
+      },
+    });
+  }
+
+  const dataSource = await getDataSource(auth, req.query.name as string);
+  if (!dataSource) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The data source you requested was not found.",
+      },
+    });
+  }
+
+  if (!dataSource.connectorProvider || !dataSource.connectorId) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_managed",
+        message:
+          "Cannot update the configuration of this Data Source because it is not managed.",
+      },
+    });
+  }
+
+  switch (dataSource.connectorProvider) {
+    case "webcrawler": {
+      if (!auth.isBuilder()) {
+        return apiError(req, res, {
+          status_code: 403,
+          api_error: {
+            type: "data_source_auth_error",
+            message:
+              "Only the users that are `builders` for the current workspace can update the configuration of this Data Source.",
+          },
+        });
+      }
+      break;
+    }
+    case "confluence":
+    case "github":
+    case "google_drive":
+    case "intercom":
+    case "notion":
+    case "slack":
+      if (!auth.isAdmin()) {
+        return apiError(req, res, {
+          status_code: 403,
+          api_error: {
+            type: "data_source_auth_error",
+            message:
+              "Only the users that are `admins` for the current workspace can update the configuration of this Data Source.",
+          },
+        });
+      }
+      break;
+
+    default:
+      assertNever(dataSource.connectorProvider);
+  }
+
+  switch (req.method) {
+    case "PATCH":
+      const parseRes = ioTsParsePayload(
+        req.body,
+        UpdateConnectorConfigurationTypeSchema
+      );
+      if (parseRes.isErr()) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body: ${parseRes.error}`,
+          },
+        });
+      }
+
+      const connectorsAPI = new ConnectorsAPI(logger);
+
+      const updateRes = await connectorsAPI.updateConfiguration({
```

Comment:
We likely want to enforce a whiltelist of configration users can update. For now `webcrawler` (to avoid allowing them to change flags as we move them to configurations)